.teamcity/_Self/templates/TestTemplatePullRequest.kt

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+package _self.templates
+
+import jetbrains.buildServer.configs.kotlin.v2019_2.*
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.commitStatusPublisher
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.PullRequests
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.pullRequests
+import jetbrains.buildServer.configs.kotlin.v2019_2.triggers.VcsTrigger
+import jetbrains.buildServer.configs.kotlin.v2019_2.triggers.vcs
+
+object TestTemplatePullRequest : TestTemplateBase() {
+    init {
```

Comment:
This inherits from the template base class, adds a trigger on pull requests, and then adds two important features.

1. "Commit status publisher" is what pushes the TeamCity status information out to GitHub.
2. "Pull requests" is a tool that improves the pull request build process: it only automates pull requests into the `main` branch (default), and it limits the builds to only people who are members or collaborators on the project. That is, some random person can't create a pull request and have it trigger a build, potentially with malicious code in it.